### PR TITLE
Added api key sanitization for additional endpoints and logs agent

### PR DESCRIPTION
**Pull request by bmikaili**
### What does this PR do?

For the `api_key` in the YAML config of the agent, whitespaces, newlines and other control characters were only stripped in some cases. This PR adds sanitization to the `api_key` in other locations, such as `logs_config` and `additional_endpoints`.

### Motivation

To prevent invalid API keys from user input.

### Describe your test plan

I added various invalid API keys to the unit tests.

### CODE REVIEW

**Review by ogaca-dd — COMMENTED** [review]
Great PR, just a few comments.

File: pkg/logs/config/config.go [hunk 1]
```diff
@@ -156,6 +158,7 @@ func buildTCPEndpoints() (*Endpoints, error) {
 	for i := 0; i < len(additionals); i++ {
 		additionals[i].UseSSL = main.UseSSL
 		additionals[i].ProxyAddress = proxyAddress
+		additionals[i].APIKey = strings.TrimSpace(additionals[i].APIKey)
```
[the remark this answers is not in the document]
Comment (bmikaili) [reply pkg/logs/config/config.go:161]: Makes sense. Changed.

File: pkg/logs/config/config.go [hunk 2]
```diff
@@ -217,6 +221,7 @@ func isSetAndNotEmpty(config coreConfig.Config, key string) bool {
 // getLogsAPIKey provides the dd api key used by the main logs agent sender.
 func getLogsAPIKey(config coreConfig.Config) string {
 	if isSetAndNotEmpty(config, "logs_config.api_key") {
+		coreConfig.SanitizeAPIKey(config, "logs_config.api_key")
```
[the remark this answers is not in the document]
Comment (bmikaili) [reply pkg/logs/config/config.go]: Done.

File: pkg/config/config.go [hunk 3]
```diff
@@ -914,9 +914,14 @@ func ResolveSecrets(config Config, origin string) error {
...
-// Avoid log ingestion breaking because of a newline in the API key
-func sanitizeAPIKey(config Config) {
-	config.Set("api_key", strings.TrimSpace(config.GetString("api_key")))
+// SanitizeAPIKey strips newlines and other control characters from a given key.
+func SanitizeAPIKey(config Config, key string) {
+	config.Set(key, Sanitize(config.GetString(key)))
+}
+
+// Sanitize strips newlines and other control characters from a given string.
+func Sanitize(s string) string {
```
Comment (ogaca-dd) [inline pkg/config/config.go:923]: `Sanitize` is quite a generic name. 
What do you think about renaming `Sanitize` to `SanitizeAPIKey` and the function line 918 can be renamed from `SanitizeAPIKey` to `SanitizeAPIKeyConfig` for example. Feel free to come with other naming.

File: pkg/config/config.go [hunk 4]
```diff
@@ -914,9 +914,14 @@ func ResolveSecrets(config Config, origin string) error {
...
+	return strings.TrimSpace(strings.Split(s, ",")[0])
```
Comment (ogaca-dd) [inline pkg/config/config.go:924]: I did not catch `strings.Split(s, ",")[0]` during the first review. 
It seems to be very specific to https://github.com/DataDog/datadog-agent/pull/5703/files#diff-21b2309ad06ba5dc719ba11f414fb0d1L419.
What do you think about handling `strings.Split(s, ",")[0]` outside this function (handle it in pkg/process/config/config.go)

**Review by ogaca-dd — APPROVED** [review]
LGTM, only a few ideas.

File: pkg/config/config.go (hunk 3 above)
Comment (bmikaili) [reply pkg/config/config.go:923]: Makes sense to me 👍

File: pkg/config/config.go (hunk 4 above)
Comment (bmikaili) [reply pkg/config/config.go:924]: Right, my bad. It is process specific.

**Review by ogaca-dd — APPROVED** [review]
Ready to 🚢!

**Pull request merged**